### PR TITLE
prevent TRT_Logger to be destroyed before TRT engine

### PR DESCRIPTION
**Pull request by Caenorst**
## Description ##
prevent TRT_Logger to be destroyed before TRT engine

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:

### Changes ###
- add TRT_Logger in TRTEngineParam
- delete TRT_Logger when TRTEngineParam

## Comments ##
- Correct a bug that is exposed when using CUDA 10.1

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: src/operator/subgraph/tensorrt/onnx_to_tensorrt.cc [hunk 1]
```diff
@@ -83,18 +83,18 @@ void PrintVersion() {
...
         const std::string& onnx_model,
         int32_t max_batch_size,
         size_t max_workspace_size,
         nvinfer1::ILogger::Severity verbosity,
         bool debug_builder) {
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 
-  TRT_Logger trt_logger(verbosity);
-  auto trt_builder = InferObject(nvinfer1::createInferBuilder(trt_logger));
+  TRT_Logger* trt_logger = new TRT_Logger(verbosity);
```
Comment (ptrendx) [inline src/operator/subgraph/tensorrt/onnx_to_tensorrt.cc]: unique_ptr?

**Comment by abhinavs95** [comment]
@Caenorst Could you have a look at the review comments?

**Comment by piyushghai** [comment]
@Caenorst Gentle ping...

**Comment by Caenorst** [comment]
Sure I can use a unique_ptr, I will make the change

**Comment by vandanavk** [comment]
@KellenSunderland for review

**Comment by karan6181** [comment]
@KellenSunderland , @ptrendx Could you please review this PR? Thanks!

**Comment by piyushghai** [comment]
@ptrendx Can you merge this PR if it looks good ?

**Pull request merged**